### PR TITLE
Line geoms

### PR DESCRIPTION
**Pull request by bethanclarke**
Changed some aes parameters to not mandatory in abline, hline and vline

### CODE REVIEW

File: instat/ucrGeom.vb [hunk 1]
```diff
@@ -181,8 +181,8 @@ Public Class ucrGeom
         'Task: add warning message to user...
         'Adding aesthetics parametters
         'Mandatory Aesthetics
-        clsgeom_abline.AddAesParameter("slope", strIncludedDataTypes:={"numeric"}, bIsMandatory:=True)
-        clsgeom_abline.AddAesParameter("intercept", strIncludedDataTypes:={"numeric"}, bIsMandatory:=True) '(required) intercept with the y axis of the line (the "b" in "y=ax+b") 
+        clsgeom_abline.AddAesParameter("slope", strIncludedDataTypes:={"numeric"})
+        clsgeom_abline.AddAesParameter("intercept", strIncludedDataTypes:={"numeric"}) '(required) intercept with the y axis of the line (the "b" in "y=ax+b")
```
Comment (maxwellfundi) [inline instat/ucrGeom.vb]: Why are these not mandatory anymore? This seems to be mandatory from this reference -http://sape.inf.usi.ch/quick-reference/ggplot2/geom_abline

**Comment by bethanclarke** [comment]
Hi Maxwell, 

The slope and intercept parameters are mandatory but it is not mandatory to have these parameters set within the `aes` function. 

We can have these parameters within `aes` function or within the `ggplot` or `geom_abline` function i.e. ``ggplot(data, aes(slope, intercept)) `+ geom_abline()` - where we set variables to these parameters 
or `ggplot(data, aes(), slope, intercept) + geom_abline()` - where we assign numeric values to these parameters 

So we have these as mandatory within in aes, then we cannot leave these blank on the dialog and then assign these parameters to a numeric value on the layer parameter tab

**Comment by dannyparsons** [comment]
This is possibly another type of "mandatory" that we should have, where a parameter is mandatory somewhere in the geom but not in a particular place.

**Comment by maxwellfundi** [comment]
@bethanclarke  Is this ready for review yet?

**Comment by bethanclarke** [comment]
Hi Maxwell, yes it's ready to review. Thanks

**Comment by maxwellfundi** [comment]
Great, @africanmathsinitiative/developers  this is ready for review.

**Comment by shadrackkibet** [comment]
Reviewing and testing this now.

File: instat/ucrGeom.vb [hunk 2]
```diff
@@ -1178,7 +1178,7 @@ Public Class ucrGeom
 
         'adding layer parameters
         'Geom_density layer parameters
-        clsgeom_violin.AddLayerParameter("draw_quantiles", "list", Chr(34) & "not(NULL)" & Chr(34), lstParameterStrings:={Chr(34) & "not(NULL)" & Chr(34), "0.25", "0.5", "0.75", "c(0.25, 0.5)", "c(0.25, 0.75)", "c(0.5,0.75)", "c(0.25,0.5,0.75)"}) 'If not(NULL) (default), draw horizontal lines at the given quantiles of the density estimate.
+        clsgeom_violin.AddLayerParameter("draw_quantiles", "list", Chr(34) & "NULL" & Chr(34), lstParameterStrings:={Chr(34) & "NULL" & Chr(34), "0.25", "0.5", "0.75", "c(0.25, 0.5)", "c(0.25, 0.75)", "c(0.5,0.75)", "c(0.25,0.5,0.75)"}) 'If not(NULL) (default), draw horizontal lines at the given quantiles of the density estimate - confusing instructions; it's say NULL is the default and when it not NULL and soemthing else then draw horizontal lines at the given quantiles of the density estimate
```
Comment (shadrackkibet) [inline instat/ucrGeom.vb]: From R documentation the default is `not(NULL)` why did you change this? . I didn't understand your comment here  perhaps you could clarify.

Comment (shadrackkibet) [reply instat/ucrGeom.vb]: ok, i see the confusion in R documentation. Should this parameter have quotes around?

Comment (shadrackkibet) [reply instat/ucrGeom.vb]: Perhaps i am doing something wrong here.I loaded survey dataset, on line plot dialog i have Fert as  X Variable and Field as single variable then on plot options layer parameters i check draw_quantiles checkbox i see the default is NULL then return back to the main dialog and click ok i found a bug.
image

**Comment by maxwellfundi** [comment]
@bethanclarke  whats the status of this?

**Comment by bethanclarke** [comment]
Hi Maxwell,

I will look at this today. Having some trouble opening data in R-Instat so haven’t been able to test the dialog yet.

Thanks

Beth

On 5 Apr 2018, at 09:45, Maxwell Fundi <notifications@github.com<mailto:notifications@github.com>> wrote:


@bethanclarke<https://github.com/bethanclarke> whats the status of this?

**Comment by shadrackkibet** [comment]
could this be same issue i am experiencing ?
When i try to load data from Instat collection i get this "error", which i don't understand why. 
image

**Comment by bethanclarke** [comment]
Hi @shadrackkibet, I spoke to Danny and he said "You may have an older version of R which has this bug. Try updating to R 3.4.4 and see if that fixes it."

This fixed my problem however, I was having a different issue to you but it's worth updating R anyway to see if it fixes the issue you're having

File: instat/ucrGeom.vb (hunk 2 above)
Comment (bethanclarke) [reply instat/ucrGeom.vb]: @shadrackkibet you are correct, NULL should not be in quotes in the code.

**Comment by dannyparsons** [comment]
@shadrackkibet which dataset did you try to load?

**Comment by shadrackkibet** [comment]
> @shadrackkibet which dataset did you try to load?

Just immediately i click **Browse**  Import from Library dialog loads and then i got above error.Additionally, when i click open from File in File menu and Open icon, i got the same issue.

**Comment by dannyparsons** [comment]
It seems like the error is in displaying the File select dialog then. I'm not getting that on mine. Is anyone else getting this?

**Comment by maxwellfundi** [comment]
Am not getting this either.

**Comment by shadrackkibet** [comment]
@dannyparsons  this is now ready for you to review.
